genomesubsampler/prototypeSelection.py

```diff
@@ -180,28 +238,32 @@ def prototype_selection_constructive_maxdist(dm, num_prototypes):
     def prototype_selection_constructive_maxdist(dm: DistanceMatrix,
     num_prototypes: int) -> List[str]:
     '''
-    if num_prototypes < 2:
-        raise ValueError(("'num_prototypes' must be >= 2, since a single "
-                          "prototype is useless."))
-    if num_prototypes >= len(dm.ids):
-        raise ValueError(("'num_prototypes' must be smaller than the number of"
-                          " elements in the distance matrix, otherwise no "
-                          "reduction is necessary."))
+    _validate_parameters(dm, num_prototypes, seedset)
 
     # initially mark all elements as uncovered, i.e. as not being a prototype
     uncovered = np.asarray([np.True_] * dm.shape[0])
+    res_set, num_found_prototypes = [], 0
+
+    if len(seedset):
+        # mark elements in the seedset as found
+        seedset = set(seedset)
```

Comment:
yeah, but you already restricted the options for the user by running you validation on the input parameters (which I think is a good idea).